src/extractor/ConfigurationExtractor.ts

```diff
@@ -1,34 +1,63 @@
 import { countBy, toPairs, maxBy, capitalize, filter } from 'lodash';
 import { fhirtypes } from 'fsh-sushi';
+import { logger } from '../utils/GoFSHLogger';
 import { ExportableConfiguration } from '../exportable';
 
 type ImplementationGuideStatus = fhirtypes.ImplementationGuideStatus;
 
 export class ConfigurationExtractor {
   static process(resources: any[]): ExportableConfiguration {
+    const igResource = ConfigurationExtractor.getIGResource(resources);
+    const missingIGProperties: string[] = [];
+    if (igResource && !igResource.url) {
+      missingIGProperties.push('url');
+    }
+    if (igResource && !igResource.fhirVersion?.length) {
+      missingIGProperties.push('fhirVersion');
+    }
+    if (missingIGProperties.length > 0) {
+      logger.warn(
+        `ImplementationGuide missing properties needed to generate configuration file: ${missingIGProperties.join(
+          ', '
+        )}`
+      );
+    }
+
     // canonical and fhirVersion are required elements in configuration, so they get defaults
-    const canonical = ConfigurationExtractor.inferCanonical(resources) || 'http://sample.org';
-    const fhirVersion = ConfigurationExtractor.inferString(resources, 'fhirVersion') || '4.0.1';
+    const canonical =
+      igResource?.url?.replace(/\/ImplementationGuide\/[^/]+$/, '') ??
+      (ConfigurationExtractor.inferCanonical(resources) || 'http://example.org');
+    const fhirVersion = igResource?.fhirVersion ?? [
+      ConfigurationExtractor.inferString(resources, 'fhirVersion') || '4.0.1'
+    ];
     const config = new ExportableConfiguration({
       canonical: canonical,
-      fhirVersion: [fhirVersion],
+      fhirVersion: fhirVersion,
       FSHOnly: true,
       applyExtensionMetadataToRoot: false
     });
-    // infer name and id using canonical
-    Object.assign(config.config, ConfigurationExtractor.inferNameAndId(canonical));
+    if (igResource) {
+      config.config.id = igResource.id;
+      config.config.name = igResource.name;
+    } else {
+      // infer name and id using canonical
+      Object.assign(config.config, ConfigurationExtractor.inferNameAndId(canonical));
+    }
     // infer status and version from most common values
-    const status = ConfigurationExtractor.inferString(
-      resources,
-      'status'
-    ) as ImplementationGuideStatus;
+    const status =
+      igResource?.status ??
+      (ConfigurationExtractor.inferString(resources, 'status') as ImplementationGuideStatus);
     if (status) {
       config.config.status = status;
     }
-    const version = ConfigurationExtractor.inferString(resources, 'version');
+    const version = igResource?.version ?? ConfigurationExtractor.inferString(resources, 'version');
     if (version) {
       config.config.version = version;
     }
```

Comment:
This also would apply to `status` and `dependencies`, but is there a reason why we can't do:
```javascript
config.config.version = igResource?.version ?? ConfigurationExtractor.inferString(resources, 'version');
```
and let the value just be not written if it is blank?

Reply:
I was mostly just following the existing pattern, but I don't see any reason why we couldn't it this way, and the code looks a little cleaner this way, so I'll update!

Reply:
I have run into a snag. By doing this, `config.config.version` and `config.config.status` become `''`, which don't get written to `sushi-config.yaml`, which is good. However, there are various `FhirToFsh` tests that compare the value of a generated config object. This means that I would now need to add in those blank values to the expected object, but the snag is that `status` has a very nicely defined type that includes `'draft' | 'active' | 'retired' | 'unknown'` but not `''`, which means TypeScript is not happy if I try to set the default configuration object in that test to have a `status: ''`. I'm leaning towards just leaving these lines how they are so we just don't include the property if it is not defined, but I wanted to check if you or anyone else had clever ideas for getting the test to pass. I could compare each individual property, but I think that actually weakens the tests a bit since a change like this wouldn't cause that type of test to break. I haven't thought of another way to fix it though.

Reply:
I'm fine with just leaving it how it is. I don't think it's worth the trouble of those changes for such a minor thing.